transaction.go

```diff
@@ -21,42 +27,118 @@ type TransactionsRequest struct {
 	Status            string
 	From              time.Time
 	To                time.Time
-	Offset            int
-	Limit             int
+	Client            *Client
+}
+
+type TransactionsIterator struct {
+	request   *TransactionsRequest
+	response  *TransactionsResponse
+	batchSize int
 }
 
 type TransactionsResponse struct {
 	Errors  []map[string]string `json:"errors"`
 	Results []Transaction       `json:"results"`
-	Pages   map[string]string   `json:"pages"`
+	Pages   Pages               `json:"pages"`
+}
+
+func (t *TransactionsRequest) GetAll() ([]Transaction, error) {
+	var resp *TransactionsResponse
+	var err error
+	if resp, err = t.get(nil); err != nil {
+		return []Transaction{}, err
+	}
+
+	return resp.Results, nil
 }
 
-func (c *Client) GetTransactions(transactionsReq TransactionsRequest) ([]Transaction, error) {
+func (t *TransactionsRequest) get(paginationParams *url.Values) (*TransactionsResponse, error) {
 	var err error
 	var req *http.Request
-	var transactions []Transaction
+	var response *TransactionsResponse
+
+	params := &url.Values{}
+	if t.CheckingAccountID != "" {
+		params.Set("checking_account_id", t.CheckingAccountID)
+	}
+	if t.Status != "" {
+		params.Set("status", t.Status)
+	}
+	dateLayout := "2006-01-02"
+	if !t.From.IsZero() {
+		params.Set("from", t.From.Format(dateLayout))
+	}
+	if !t.To.IsZero() {
+		params.Set("to", t.To.Format(dateLayout))
+	}
+
+	var url *url.URL
+	if url, err = url.Parse(fmt.Sprintf("%s/%s", ApiBase, "transactions/")); err != nil {
+		return nil, err
+	}
+
+	if paginationParams != nil {
+		url.RawQuery = fmt.Sprintf("%s&%s", params.Encode(), paginationParams.Encode())
+	}
 
-	if req, err = http.NewRequest("GET", fmt.Sprintf("%s/%s", ApiBase, "transactions/"), nil); err != nil {
-		return transactions, err
+	if req, err = http.NewRequest("GET", url.String(), nil); err != nil {
+		return nil, err
 	}
 	var resp *http.Response
 
-	if resp, err = c.Do(req); err != nil {
-		return transactions, err
+	if resp, err = t.Client.do(req); err != nil {
+		return nil, err
 	}
 
-	var response TransactionsResponse
-	if err = json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return transactions, err
+	if err = json.NewDecoder(resp.Body).Decode(response); err != nil {
+		return nil, err
 	}
+	return response, nil
+}
 
-	return transactions, nil
+func (r *TransactionsRequest) Iterator() TransactionsIterator {
+	return TransactionsIterator{
+		request:   r,
+		batchSize: MaxBatchSize,
+	}
 }
 
-func (c *Client) Do(req *http.Request) (*http.Response, error) {
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.accessToken))
-	if c.clientVersion != "" {
-		req.Header.Set("Client-Version-Id", c.clientVersion)
+func (t *TransactionsIterator) SetBatchSize(n int) {
+	if n < MaxBatchSize {
+		t.batchSize = n
+	}
+}
+
+func (t *TransactionsIterator) Next() ([]Transaction, []map[string]string, error) {
```

Comment:
I think returning `([]Transaction, error)` makes sense, and including errors in the `error` object, since normally we shouldn't run into errors in the normal case; but it's up to you.

My reasoning is that errors being returned from the API shouldn't happen all that much, and so when they do (rarely) happen it's probably fine to have it stuffed in as part of `error`; and that way the API would be easier to use.

So when it's called, they would do:
```Golang
txs, err := Next()
if err != nil {
  // ...
}
```

... as opposed to needing to check both types of errors, and making sure that they were both empty, which would make it a little more verbose